xonsh/completer.py

```diff
@@ -124,60 +166,52 @@ def complete_from_context(self, completion_context, old_completer_args=None):
                 else:
                     lprefix = 0
 
-            if res is None or len(res) == 0:
+            if res is None:
                 continue
 
+            items = []
+            for comp in res:
+                items.append(comp)
+                yield Completer._format_completion(
+                    comp,
+                    completion_context,
+                    completing_contextual_command,
+                    lprefix or 0,
+                    custom_lprefix,
+                )
+
             if trace:
                 print(
-                    f"TRACE COMPLETIONS: Got {len(res)} results"
+                    f"TRACE COMPLETIONS: Got {len(items)} results"
                     f" from {'' if is_exclusive_completer(func) else 'non-'}exclusive completer '{name}':"
                 )
-                sys.displayhook(res)
-
-            if (
-                completing_contextual_command
-                and completion_context.command.is_after_closing_quote
-            ):
-                """
-                The cursor is appending to a closed string literal, i.e. cursor at the end of ``ls "/usr/"``.
-                1. The closing quote will be appended to all completions.
-                    I.e the completion ``/usr/bin`` will turn into ``/usr/bin"``
-                    To prevent this behavior, a completer can return a ``RichCompletion`` with ``append_closing_quote=False``.
-                2. If not specified, lprefix will cover the closing prefix.
-                    I.e for ``ls "/usr/"``, the default lprefix will be 6 to include the closing quote.
-                    To prevent this behavior, a completer can return a different lprefix or specify it inside ``RichCompletion``.
-                """
-                closing_quote = completion_context.command.closing_quote
-                if not custom_lprefix:
-                    lprefix += len(closing_quote)
-
-                def append_closing_quote(completion: Completion):
-                    if isinstance(completion, RichCompletion):
-                        if completion.append_closing_quote:
-                            return completion.replace(
-                                value=completion.value + closing_quote
-                            )
-                        return completion
-                    return completion + closing_quote
-
-                res = map(append_closing_quote, res)
-
-            completions.update(apply_lprefix(res, lprefix))
+                sys.displayhook(items)
+
+            if not items:  # empty completion
```

Comment:
This check needs to be before tracing so that we won't trace an empty result

Reply:
But sometimes one might want to see that their functions returned empty results right?

Reply:
Not really. You can see which completers are registered with `completer list`
The tracing helps you to understand what really happened in runtime so these prints are just getting in the way imo

Reply:
Having it registered and the function returning empty results are different I think. Why do you wanr skip the comment on empty set?

Reply:
Because all completers are run until a non exclusive one returns something.
So it's clear and known what completers will run, and there's no need to print their empty results
I think it just clutters the trace